thunder/tests/opinfos.py

```diff
@@ -6152,6 +6152,32 @@ def sample_generator(op, device, dtype, requires_grad, **kwargs):
 nn_ops.append(max_pool3d_opinfo)
 
 
+def one_hot_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=torch.long, requires_grad=requires_grad)
+
+    test_shapes = [
+        (10,),
+        (5, 10),
+        (3, 5, 10),
+    ]
+
+    max_value = 9
+    for shape in test_shapes:
+        for num_classes in range(1, max_value + 1):
+            a = make(shape, low=0, high=num_classes - 1)  # use non-negative integers
+
+            yield SampleInput(a, num_classes=num_classes)
+
+
+one_hot_opinfo = OpInfo(
+    ltorch.one_hot,
+    sample_input_generator=one_hot_sample_generator,
+    torch_reference=torch.nn.functional.one_hot,
+    dtypes=(datatypes.signedinteger,),
```

Comment:
Does PyTorch actually support dtypes other than int64? In the generator the dtype argument is ignored, so all the generated tests will run with int64 tensors, even though they claim to be testing other signed integer types.

Reply:
Reviewing the rest of the PR, maybe this should only generate tests for additional signed integers when using the nvFuser executor? If that's the case then dtypes other than int64 can be skipped when using the torch executor. 

Here's an example of an executor-specific test directive:

https://github.com/shaharelys/lightning-thunder/blob/9ff2c62491e52d580d8acfead46e15b40773fa1e/thunder/tests/opinfos.py#L789

Reply:
Not sure this is resolved.